src/index.spec.ts

```diff
@@ -4,9 +4,233 @@ import {
   removeDuplicatesFromArray,
   createRandomProduct,
   getStarWarsPlanets,
-  createProduct 
+  createProduct,
 } from './index';
 
-test('Testing time!', () => {
-  console.log('welcome');
+import cases from 'jest-in-case';
+
+import fetch from 'node-fetch';
+import { mocked } from 'ts-jest/utils';
+
+function casify(obj: unknown) {
+  return Object.entries(obj).map(([key, value]) => ({
+    name: `${
+      value && value.hasOwnProperty('input') ? value.input : value
+    } - ${key}`,
+    value,
+  }));
+}
+
+/* is Integer */
+
+cases(
+  'is an integer?: valid input',
+  ({ value }) => {
+    expect(isInteger(value)).toBe(true);
+  },
+  casify({
+    'positive integer': 1234,
+    'negative integer': -1234,
+    zero: 0,
+  }),
+);
+
+cases(
+  'is an integer?: invalid input',
+  ({ value }) => {
+    expect(isInteger(value)).toBe(false);
+  },
+  casify({
+    decimal: 0.5,
+    string: 'n1234',
+    null: null,
+    object: {},
+  }),
+);
+
+/* toLowerCase */
+
+cases(
+  "toLowerCase: invalid input returns 'Please provide a string' message",
+  ({ value }) => {
+    expect(toLowerCase(value)).toBe('Please provide a string');
+  },
+  casify({
+    'empty string': '',
+    null: null,
+    //object: {}, // fails
+  }),
+);
+
+cases(
+  'toLowerCase: valid input',
+  ({ value }) => {
+    expect(toLowerCase(value.input)).toBe(value.expected);
+  },
+  casify({
+    'first capital': {
+      input: 'Hola',
+      expected: 'hola',
+    },
+    'all capitals': {
+      input: 'HOLA',
+      expected: 'hola',
+    },
+    'all smallcaps': {
+      input: 'hola',
+      expected: 'hola',
+    },
+    numbers: {
+      input: '1234',
+      expected: '1234',
+    },
+    alphanumeric: {
+      input: 'A1234',
+      expected: 'a1234',
+    },
+  }),
+);
+
+/* removeDuplicatesFromArray */
+
+cases(
+  'removeDuplicatesFromArray: valid input',
+  ({ value }) => {
+    expect(removeDuplicatesFromArray(value.input)).toEqual(value.expected);
+  },
+  casify({
+    'no duplicates alphanum': {
+      input: ['a', 'b', 3],
+      expected: ['a', 'b', 3],
+    },
+    'no duplicates numeric': {
+      input: [1, 2, 3],
+      expected: [1, 2, 3],
+    },
+    'no duplicates alpha': {
+      input: ['a', 'b', 'c'],
+      expected: ['a', 'b', 'c'],
+    },
+    'one element': {
+      input: [1],
+      expected: [1],
+    },
+    empty: {
+      input: [],
+      expected: [],
+    },
+  }),
+);
+
+cases(
+  'removeDuplicatesFromArray: invalid input throws Error',
+  ({ value }) => {
+    const badInput = () => {
+      removeDuplicatesFromArray(value);
+    };
+
+    expect(badInput).toThrowError(
+      'please provide an array of numbers or strings',
+    );
+  },
+  casify({
+    null: null,
+    object: {},
+    'single value': 1234,
+  }),
+);
+
+/* createProduct */
+
+describe('createProduct', () => {
+  test('createProduct returns a valid Product', () => {
```

Comment:
We should use the format _it should <our expected result>_ to describe the test.
You can use `test('it should ...')` or `it('should ...')`